src/main/java/net/minecraftforge/fmp/client/multipart/IWeightedBakedModel.java

```diff
@@ -0,0 +1,104 @@
+package net.minecraftforge.fmp.client.multipart;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
+import net.minecraft.client.renderer.block.model.ItemOverrideList;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.util.EnumFacing;
+
+/**
+ * Interface that represents a model whose sides have different sorting priorities. Lower weights get added after higher
+ * weights, rendering on top of them and vice versa.
+ */
+public interface IWeightedBakedModel extends IBakedModel
+{
+    
+    /**
+     * Gets the weight of the quads on a specific side for the specified state.
+     */
+    public int getWeight(@Nullable IBlockState state, @Nullable EnumFacing side, long rand);
+
+    public static class WeightedModelAdapter implements IWeightedBakedModel
+    {
```

Comment:
You're mixing implementations and api what? How is this suposed to be used? Should people care about IModelWeightProvider?